neard/src/cli.rs

```diff
@@ -409,6 +419,14 @@ impl RunCmd {
         let (tx, rx) = oneshot::channel::<()>();
         let sys = actix::System::new();
         sys.block_on(async move {
+            // `SIGHUP` lets user inform the process that the logging config has changed and
+            // EnvFilter needs to be reinitialized.
+            if cfg!(unix) {
+                if let Some(ref log_config) = &near_config.config.log_config {
+                    spawn_log_config_watcher(home_dir.join(log_config));
+                };
+            }
+
```

Comment:
How about putting it with handling of the other signals?

```rust
            let sig = if cfg!(unix) {
                use tokio::signal::unix::{signal, SignalKind};
                let mut sigint = signal(SignalKind::interrupt()).unwrap();
                let mut sigterm = signal(SignalKind::terminate()).unwrap();
                let mut sighup = signal(SignalKind::hangup()).unwrap();
                loop {
                    futures::select! {
                        _ = sighup.recv().fuse() => {
                            /* handle SIGHUP */
                        },

                        _ = sigint .recv().fuse() => break "SIGINT",
                        _ = sigterm.recv().fuse() => break "SIGTERM",
                        _ = rx.fuse() => break "ClientActor died",
                    }
                }
            } else {
                // TODO(#6372): Support graceful shutdown on windows.
                tokio::signal::ctrl_c().await.unwrap();
                "Ctrl+C"
            };
```

Reply:
Managed to do it, but the solution is far from elegant.

Reply:
Hmm… I hate to say it after you’ve spent time figuring out how to make it work but I think the old version might be better after all. Either works though. Having centralised place where signals are handled might be useful in the future.

Reply:
The old version didn't work either:
- set a SIGHUP handler in a separate thread.
- SIGINT or repeated SIGHUP are handled
- A SIGINT after a SIGHUP doesn't get handled. I couldn't figure out why. According to the docs it should be handled.

Reply:
Resolved with @matklad 's help.